src/Widgets/TaskFormRevealer.vala

```diff
@@ -0,0 +1,173 @@
+
+    public signal void cancel_clicked ();
+    public signal void save_clicked (ECal.Component task);
+    public signal void delete_clicked (ECal.Component task);
+
+    public TaskFormRevealer (ECal.Component task) {
+        Object (task: task);
+    }
+
+    construct {
+        due_switch = new Gtk.Switch ();
+        due_switch.valign = Gtk.Align.CENTER;
+
+        var due_label = new Gtk.Label (_("Schedule:"));
+        due_datepicker = new Granite.Widgets.DatePicker ();
+        due_timepicker = new Granite.Widgets.TimePicker ();
+
+        var due_datetimepicker = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 12);
+        due_datetimepicker.add (due_label);
+        due_datetimepicker.add (due_switch);
+        due_datetimepicker.add (due_datepicker);
+        due_datetimepicker.add (due_timepicker);
+
+        var description_textview = new Gtk.TextView ();
+        description_textview.margin = 12;
+        description_textview.top_margin = description_textview.bottom_margin = 12;
+        description_textview.set_wrap_mode (Gtk.WrapMode.WORD_CHAR);
+        description_textview.accepts_tab = false;
+
+        description_textbuffer = new Gtk.TextBuffer (null);
+        description_textview.set_buffer (description_textbuffer);
+
+        var description_scrolled_window = new Gtk.ScrolledWindow (null, null);
+        description_scrolled_window.hscrollbar_policy = Gtk.PolicyType.EXTERNAL;
+        description_scrolled_window.height_request = 140;
+        description_scrolled_window.add (description_textview);
+
+        var description_frame = new Gtk.Frame (null);
+        description_frame.margin_bottom = 0;
+        description_frame.add (description_scrolled_window);
+
+        var delete_button = new Gtk.Button ();
+        delete_button.sensitive = false;
+        delete_button.label = _("Delete Task");
+        delete_button.halign = Gtk.Align.START;
+        delete_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+
+        var cancel_button = new Gtk.Button ();
+        cancel_button.label = _("Cancel");
+        cancel_button.halign = Gtk.Align.START;
+
+        var save_button = new Gtk.Button ();
+        save_button.label = _("Save Changes");
+        save_button.halign = Gtk.Align.END;
+        save_button.get_style_context ().add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
+
+        var button_box = new Gtk.ButtonBox (Gtk.Orientation.HORIZONTAL);
+        button_box.baseline_position = Gtk.BaselinePosition.CENTER;
+        button_box.set_layout (Gtk.ButtonBoxStyle.END);
+
+        button_box.add (delete_button);
+        button_box.set_child_secondary (delete_button, true);
+        button_box.set_child_non_homogeneous (delete_button, true);
+        button_box.add (cancel_button);
+        button_box.set_child_non_homogeneous (cancel_button, true);
+        button_box.add (save_button);
+
+        var box = new Gtk.Box (Gtk.Orientation.VERTICAL, 12);
```

Comment:
Instead of two boxes, I would recommend Gtk.Grid

Reply:
Tried to get use one `Gtk.Grid` instead of the two `Gtk.Box`, but I was not able to rebuild the interface using the Grid: The due fields are then spoiled over the whole width as soon as I set `description_frame.vexpand = true`. Additionally setting `grid.column_homogeneous = false` and the due field's `.halign =  Gtk.Align.START` did not help either.

Whats the approach to recreate the given interface with only one `Gtk.Grid`? And is it in general advised to use `Gtk.Grid` over nesting `Gtk.Boxes`? What are the drawbacks of nesting Widgets?

At the moment I feel like nesting boxes is a lot easier to wrap ones head around - but maybe that's just me, lacking the experience using `Gtk.Grid`.

Reply:
@danrabbit how shall we proceed with the `Gtk.Grid` / `Gtk.Box` thingy?